src/Component/Repository/RepositoryInterface.php

```diff
@@ -16,6 +16,10 @@
 use Doctrine\Persistence\ObjectRepository;
 use Sylius\Component\Resource\Model\ResourceInterface;
 
+/**
+ * @template T
+ * @extends ObjectRepository<T>
+ */
 interface RepositoryInterface extends ObjectRepository
```

Comment:
I'd add the generic annotations to the methods in this interface too.

`createPaginator()` returns `iterable<T>`, and the parameter for `add()` and `remove()` is `T`

Reply:
Yeah, good idea. Thanks.